compiler/damlc/daml-compiler/src/DA/Daml/Compiler/DataDependencies.hs

```diff
@@ -193,94 +199,94 @@ generateSrcFromLf env = noLoc mod
             { ideclExt = noExt
             , ideclSourceSrc = NoSourceText
             , ideclName =
-                  noLoc $ mkModuleName $ T.unpack $ LF.moduleNameString modRef
-            , ideclPkgQual =
-              if isStable
-                then Nothing -- we don’t do package qualified imports for modules that should come from the current SDK.
-                else Just $ StringLiteral NoSourceText $ mkFastString $
-                  -- Package qualified imports for the current package need to use "this" instead
-                  -- of the package id.
-                  if refUnitId == unitId then "this" else fst (splitUnitId $ unitIdString refUnitId)
+                  noLoc $ mkModuleName $ T.unpack $ LF.moduleNameString modRefModule
+            , ideclPkgQual = do
+                guard modRefIsStable -- we don’t do package qualified imports
+                    -- for modules that should come from the current SDK.
+                Just $ StringLiteral NoSourceText $ mkFastString $
+                    -- Package qualified imports for the current package
+                    -- need to use "this" instead of the package id.
+                    if modRefUnitId == unitId
+                        then "this"
+                        else fst (splitUnitId $ unitIdString modRefUnitId)
             , ideclSource = False
             , ideclSafe = False
             , ideclImplicit = False
             , ideclQualified = True
             , ideclAs = Nothing
             , ideclHiding = Nothing
             } :: LImportDecl GhcPs
-        | (isStable, refUnitId, modRef) <- modRefs
+        | ModRef{..} <- Set.toList modRefs
          -- don’t import ourselves
-        , not (modRef == lfModName && unitId == unitId)
+        , not (modRefModule == lfModName && modRefUnitId == unitId)
```

Comment:
I *think* this is an unrelated bugfix.

Reply:
This really should be in HLint :slightly_smiling_face: